python/pyspark/serializers.py

```diff
@@ -295,7 +297,33 @@ def create_array(s, t):
             raise RuntimeError(error_msg % (s.dtype, t), e)
         return array
 
-    arrs = [create_array(s, t) for s, t in series]
+    arrs = []
+    for s, t in series:
+        if t is not None and pa.types.is_struct(t):
+            if not isinstance(s, pd.DataFrame):
+                raise ValueError("A field of type StructType expects a pandas.DataFrame, " +
```

Comment:
(not a big deal but ` +` at the end wouldn't be needed)

Reply:
Oh right, thanks!